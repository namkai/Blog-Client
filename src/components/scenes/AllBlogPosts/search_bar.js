import React from 'react';

export const SearchBar = ({ query, updateQuery }) =>
	<li className="media list-group-item p-4">
		<div className="input-group">
			<input
				type="text"
				placeholder="Search for a blog post"
				onChange={e => updateQuery(e.target.value)}
				value={query}
				className="form-control"
			/>
		</div>
	</li>;

export default SearchBar;
