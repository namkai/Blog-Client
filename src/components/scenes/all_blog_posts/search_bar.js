import PropTypes from 'prop-types';
import React from 'react';

export const SearchBar = ({ query, updateQuery }) =>
	<li className="media list-group-item p-4">
		<div className="input-group">
			<input
				type="text"
				placeholder="Search for a blog post"
				onChange={e => updateQuery(e.target.value)}
				value={query}
				className="form-control"
			/>
		</div>
	</li>;

SearchBar.defaultProps = {
	query: '',
	updateQuery: PropTypes.func,
};

SearchBar.propTypes = {
	query: PropTypes.string,
	updateQuery: PropTypes.func,
};

export default SearchBar;
