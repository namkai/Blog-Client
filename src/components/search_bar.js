import React from 'react';
import { connect } from 'react-redux';
import { updateQuery } from '../actions/query_actions';

export const Searchbar = ({ query, updateQuery }) =>
	<li className="media list-group-item p-4">
		<div className="input-group">
			<input
				type="text"
				placeholder="Search for a blog post"
				onChange={e => updateQuery(e.target.value)}
				value={query}
				className="form-control"
			/>
		</div>
	</li>;

export default connect(({ query }) => ({ query }), { updateQuery })(Searchbar);
