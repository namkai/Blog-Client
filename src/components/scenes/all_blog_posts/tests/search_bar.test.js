import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { SearchBar } from '../search_bar';

it('renders correctly', () => {
	const searchbar = shallow(
		<SearchBar />
	);
	expect(toJson(searchbar)).toMatchSnapshot();
});