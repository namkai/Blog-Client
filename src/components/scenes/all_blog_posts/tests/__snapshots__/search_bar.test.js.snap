// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="media list-group-item p-4"
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search for a blog post"
      type="text"
      value=""
    />
  </div>
</li>
`;
